api/v1alpha4/cluster_types.go

```diff
@@ -60,6 +60,72 @@ type ClusterSpec struct {
 	// for provisioning infrastructure for a cluster in said provider.
 	// +optional
 	InfrastructureRef *corev1.ObjectReference `json:"infrastructureRef,omitempty"`
+
+	// This encapsulates the topology for the cluster.
+	// NOTE: This feature is alpha; it is required to enable the ClusterTopology
+	// feature gate flag to activate managed topologies support.
+	// +optional
+	Topology *Topology `json:"topology,omitempty"`
+}
+
+// Topology encapsulates the information of the managed resources.
+type Topology struct {
+	// The name of the ClusterClass object to create the topology.
+	Class string `json:"class"`
+
+	// The kubernetes version of the cluster.
+	Version string `json:"version"`
+
+	// RolloutAfter performs a rollout of the entire cluster one component at a time,
+	// control plane first and then machine deployments.
+	// +optional
+	RolloutAfter *metav1.Time `json:"rolloutAfter,omitempty"`
+
+	// The information for the Control plane of the cluster.
+	ControlPlane ControlPlaneTopology `json:"controlPlane"`
+
+	// Workers encapsulates the different constructs that form the worker nodes
+	// for the cluster.
+	// +optional
+	Workers *WorkersTopology `json:"workers,omitempty"`
+}
+
+// ControlPlaneTopology specifies the parameters for the control plane nodes in the cluster.
+type ControlPlaneTopology struct {
+	Metadata ObjectMeta `json:"metadata,omitempty"`
+
+	// The number of control plane nodes.
```

Comment:
```suggestion
	// Replicas is the number of control plane nodes.
```

side note: we should enable comment lint